### ISSUE TITLE
Social plugin: runtime error with generated files

### ISSUE THREAD

**Issue by thorstenpeter-sophos** [squidfunk/mkdocs-material#7520]
### Bug description

When using the Insiders version of the `Social` plugin in combination with another plugin that creates generated files, it raises an `AttributeError` exception.

Generated files were added with mkdocs 1.6 and don't have an `abs_scr_path` attribute.

### Related links

- [MkDocs release notes: Enabling true generated files and expanding the File API](https://www.mkdocs.org/about/release-notes/#enabling-true-generated-files-and-expanding-the-file-api)
- [Discussion: Social plugin: runtime error with virtual files](https://github.com/squidfunk/mkdocs-material/discussions/7517)

### Reproduction

[9.5.34+insiders.4.53.12-social-virtual-files-error.zip](https://github.com/user-attachments/files/16978113/9.5.34%2Binsiders.4.53.12-social-virtual-files-error.zip)

The `foo` plugin used in the reproduction is defined as follows:

```
from mkdocs.plugins import BasePlugin
from mkdocs.structure.files import File, Files, InclusionLevel

class FooPlugin(BasePlugin):

    def on_files(self, files: Files, **kwargs):
        return files.append(File.generated(
            config=kwargs.get("config"),
            src_uri="foo.md",
            content="",
            inclusion=InclusionLevel.NOT_IN_NAV,
        ))
```

### Steps to reproduce

Run `mkdocs build` in the reproduction.

The build will fail with the following exception:

```
File "C:\git\mkdocs_material_insiders_minimal_reproduction\venv\Lib\site-packages\mkdocs\plugins.py", line 566, in run_event
  result = method(item, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^
File "C:\git\mkdocs_material_insiders_minimal_reproduction\venv\Lib\site-packages\material\plugins\social\plugin.py", line 180, in on_files
  if file.abs_src_path.startswith(_templates_dirpath()):
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'startswith'
```

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by thorstenpeter-sophos** [comment]
This zip contains the `foo` plugin used in the reproduction.

[foo-0.0.1-py3-none-any.zip](https://github.com/user-attachments/files/16978998/foo-0.0.1-py3-none-any.zip)

**Comment by squidfunk** [comment]
Judging from [a comment](https://github.com/mkdocs/mkdocs/blob/bb7e8b62185b11d9f59bb7f50b13c15134f62f8a/mkdocs/structure/files.py#L200-L206) in the `File` class, `abs_src_path` was made optional in MkDocs 1.6.0:

```
*   Since MkDocs 1.6 a file may alternatively be stored in memory - `content_string`/`content_bytes`.

        Then `src_dir` and `abs_src_path` will remain `None`. `content_bytes`/`content_string` need
        to be written to, or populated through the `content` argument in the constructor.

        But `src_uri` is still populated for such files as well! The virtual file pretends as if it
        originated from that path in the `docs` directory, and other values are derived.
```

Well, so this is essentially a change in behavior that I would definitely consider to be a breaking change. Yet, it's not MkDocs 2.0 but 1.6. This might impact a lot of plugins, since up to now `abs_src_path` was always available. Sigh. Let's fix this one and see whether other plugins suffer from the same problem as we go.

**Comment by squidfunk** [comment]
Fixed in [ee8d243f9](https://github.com/squidfunk/mkdocs-material-insiders/commit/ee8d243f989a46ff8fe391c5f8245650dede8839).

**Comment by squidfunk** [comment]
Released as part of [9.5.34+insiders-4.53.13](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.53.13).
